### PR TITLE
TariikMVP/FEATURE/ProjectsTable

### PR DESCRIPTION
**Pull request by TarikVelic**
--------------------------------------------------------------------
 MAIN
 * Custom Table creation --------------------------------------------------------------------
 OTHER
 * <other msg> --------------------------------------------------------------------

### CODE REVIEW

**Comment by becarselman** [comment]
*components_(component folder in src project folder)
                         |
                          * Login_(main specific component folder)
                                         |
                                         View.js. (main file of component)
                                         index.js (connect Container and View)
                                         Container.js ( file for mapStateToProps...)
                                         *modules_(everything regarding state management goes in this folder and it is just for this 
                                                            component. Of course you can reuse in other modules file for example action)
                                                            |
                                                            reducer.js
                                                            saga.js
                                                            actions.js
                                                            selectors.js
                                                            types.js
                                         *components_
                                                                  |
                                                                  LoginForm.js
                                                                  columns.js
                                                                  UserItem.js
                                                                   (all other components that are included in Login for example itemCard that will be 
                                                                      mapped)

**Comment by becarselman** [comment]
image
* In src folder we have some other folders for example fonts, images, utils and components
image
*Component folder is main folder for developing application.
in this folder we create folder for each component and that gives us power to have separate system for each component but with possibility to reuse everything.

image
* Now single component folder will look like this. We have View.js file as main file, component scss file Container file for importing stuff from state and actions and index.js file for connecting view and container.
image
*In modules folder we put everything regarding state management for that component. For example actions for getChildren, setChildren... and sagas for that actions ....
*In component folder we will put helper components for this Children component (childrenItem, columns...)


If some component can be reusable in more screens we will create separate folder for them for example this Tables folder
image



Please use this structure in your project and your life will get easier when this app get bigger. @TarikVelic @ElvaRivle @meliha01

File: src/components/Table/CustomTable.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  const handleLastPage = () => {
+    setCurrentPage(Math.ceil(totalItems / pageSize));
+  };
+
+  return (
+    <div className="table-container">
+      <Table dataSource={pageData} columns={columns} className="table" pagination={false} />
+
+      <div className="pagination-container">
```
Comment (becarselman) [inline src/components/Table/CustomTable.js]: You can move this whole part in separate component in Tables/components folder
You can call them something like noOfRows or something like that

File: src/components/Table/CustomTable.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        <div className="pagination-info">
+          <span className="rows-text">Rows per page:</span>
+          <Select value={pageSize.toString()} onChange={handlePageSizeChange}>
+            {[1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((value) => (
```
Comment (becarselman) [inline src/components/Table/CustomTable.js]: I think 1-10 is bad choice. I will go with 10 20 50. and also move this array outside of jsx ( connecting to previous comment you will move this outside of jsx in separate component noOfRows)

File: src/components/Table/CustomTable.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+              <Select.Option key={value} value={value.toString()}>
```
Comment (becarselman) [inline src/components/Table/CustomTable.js]: I see you use second time value.toString. Why you don't write initial value in array like string "10", "20" ...

File: src/components/Table/CustomTable.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                {value}
+              </Select.Option>
+            ))}
+          </Select>
+          <span className="show-total">{paginationConfig.showTotal(totalItems, [startIndex + 1, endIndex])}</span>
+        </div>
+        <div className="pagination-right">
+        <Pagination {...paginationConfig} className="custom-pagination" showQuickJumper={false} />
+        <div className="next-last-buttons">
+          <button onClick={handleNextPage} disabled={currentPage === Math.ceil(totalItems / pageSize)}>
```
Comment (becarselman) [inline src/components/Table/CustomTable.js]: move this condition outside of jsx

File: src/components/Table/CustomTable.scss [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    @include gilroy-medium();
+    color: $gray-color;
+  }
+}
+
+.custom-pagination {
+  display: flex;
+  align-items: center;
+
+  .ant-pagination-prev,
```
Comment (becarselman) [inline src/components/Table/CustomTable.scss]: here you have example how to change antd arrows for prev and next page https://codesandbox.io/s/pagination-settings-antd-4-21-6-forked-0422vs?file=/demo.js

**Review by becarselman — CHANGES_REQUESTED** [review]

File: src/components/Table/CustomTable.scss [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+  .pagination-container {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    width: 100%;
+    margin-top: 16px;
+
+    .pagination-info {
+      @include gilroy-medium();
```
Comment (becarselman) [inline src/components/Table/CustomTable.scss]: why you create mixin as function here.

File: src/components/Table/CustomTable.scss [hunk 7]
```diff
@@ -0,0 +1,160 @@
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;500&display=swap');
```
Comment (becarselman) [inline src/components/Table/CustomTable.scss]: you don't need to import font here if you are using one from mixin

File: src/components/Table/CustomTable.scss [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+      align-items: center;
+    }
+    .ant-pagination-total-text {
+      display: none;
+    }
+  }
+}
+
+.ant-pagination-next, 
+.ant-pagination-prev {
```
Comment (becarselman) [inline src/components/Table/CustomTable.scss]: I don't like this solution

File: src/components/Table/CustomTable.scss [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    color: $gray-color;
+  }
+}
+
+.custom-pagination {
+  display: flex;
+  align-items: center;
+
+  .ant-pagination-item-active {
+    border-color: $gray-color !important;
```
Comment (becarselman) [inline src/components/Table/CustomTable.scss:81]: try to avoid !important

File: src/components/Table/Pagination.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+  const handleLastPage = () => {
+    onPageChange(Math.ceil(totalItems / pageSize));
+  };
+
+  const isLastPage = currentPage === Math.ceil(totalItems / pageSize);
+
+  const pageSizeSelect = (
+    <Select value={pageSize.toString()} onChange={handlePageSizeChange}>
+      {['5', '10', '20'].map((value) => (
```
Comment (becarselman) [inline src/components/Table/Pagination.js]: One more time you ignore my comment. 1. move map outside of jsx
2. move array in separate variable. 3. use 5 10 and 20 entries is small amount of data, use 10 20 50.

File: src/components/projects/components/data.js [hunk 11]
```diff
@@ -0,0 +1,288 @@
+export const tableData = [
+    {
+      key: '1',
```
Comment (becarselman) [inline src/components/projects/components/data.js]: you can use chatgtp to create this data. or you can create unique key by your self

**Review by becarselman — CHANGES_REQUESTED** [review]

File: package.json [hunk 12]
```diff
@@ -4,11 +4,14 @@
...
   "dependencies": {
     "@ant-design/icons": "^5.0.1",
+    "@reduxjs/toolkit": "^1.9.5",
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
+    "antd": "^5.5.1",
     "axios": "^1.3.5",
     "cors": "^2.8.5",
+    "numeral": "^2.0.6",
```
Comment (AntMarkoLovric) [inline package.json:15]: Do we need to import this library or we can format it another way?

File: src/components/Table/CustomTable.js [hunk 13]
```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect } from 'react';
+import { Table, Spin } from 'antd';
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.js]: Please format import as we discussed in the previous PR
-React import
-Third party import
-Local import
-Images
-SCSS files

File: src/components/Table/CustomTable.js [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+  const handlePageSizeChange = (value) => {
+    onPageChange(1);
+    onPageSizeChange(parseInt(value));
+  };
+
+  const handlePageChange = (page) => {
+    onPageChange(page);
+  };
+
+  useEffect(() => {
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.js:39]: why do you use this in the use effect block, can we implement onClick function which will change value of the filter in state

File: src/components/Table/CustomTable.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+    const filteredData = data.filter((item) => {
+      return item.status === selectedNavLabel || selectedNavLabel === 'All Projects';
+    });
+    setPageData(filteredData);
+  }, [data, selectedNavLabel]);
+
+  const handleSearch = (input) => {
+    onSearchChange(input);
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.js]: i think it is better to save these state values in redux state instead of component state (useState). Then we can use it and access it throughout the whole project

File: src/components/Table/CustomTable.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+  const tableHeader = (
+    <TableHeader totalCount={totalCount} handleSearch={handleSearch} />
+  );
+
+  return (
+    <div className="custom-table-container">
+      <Navbar
+        navLabels={navLabels} 
+        handlePageSelect={(page, label) => {
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.js]: move inline functions from jsx

File: src/components/Table/CustomTable.js [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+          onNavSelect(label);
+        }}
+      />
+      <div className="table-container">
+        {isLoading ? (
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.js]: move ternary operator from jsx

File: src/components/Table/CustomTable.scss [hunk 18]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  .ant-pagination-item-active {
+    border-color: $gray-color !important;
+    border-radius: 4px;
+    background-color: $border-color;
+
+    a {
+      color: $black-color;
+      background-color: #F5F6FB;
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.scss]: move this color into colors file or reuse it

File: src/components/Table/components/Navbar.js [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+
+import NavLink from './Navlink';
+
+const Navbar = ({ navLabels, handlePageSelect }) => {
+  const [selectedPage, setSelectedPage] = useState(1);
+
+  return (
+    <div className="navbar-container">
+      <div className="navbar-links">
+        {navLabels.map((label, index) => (
```
Comment (AntMarkoLovric) [inline src/components/Table/components/Navbar.js]: move map outside of jsx

File: src/components/Table/components/Navbar.js [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+          <NavLink
+            key={index}
+            label={label}
+            onClick={() => {
+              handlePageSelect(index + 1, label);
```
Comment (AntMarkoLovric) [inline src/components/Table/components/Navbar.js]: also move on click function outside of map, it will allow this on click function to be remembered, and it will be called when necessary, and not generated every time when you render navbar

File: src/components/dashboard/Dashboard.js
Comment (AntMarkoLovric) [inline src/components/dashboard/Dashboard.js:1]: when you finish testing and working on feature, remove changes on files that are not in your task (Dashboard, Sidebar, Routing etc.)

File: src/components/projects/View.js [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+import './Projects.scss';
+import CustomTable from '../Table/CustomTable';
+import { tableColumns } from './components/columns';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchProjectsRequest } from './modules/actions';
+
+const Projects = () => {
+  const dispatch = useDispatch();
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
```
Comment (AntMarkoLovric) [inline src/components/projects/View.js]: I think if you move state values in redux state from one of previous comments, you can access here and also you can use actions for that

File: src/components/projects/View.js [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+  const [selectedProjectStatus, setSelectedProjectStatus] = useState('All Projects');
+  const [searchInput, setSearchInput] = useState('');
+  const navLabels = ['All Projects', 'Active', 'Inactive', 'Completed'];
+
+  useEffect(() => {
+    dispatch(fetchProjectsRequest(page, pageSize, selectedProjectStatus, searchInput));
```
Comment (AntMarkoLovric) [inline src/components/projects/View.js]: move this to container file and trigger actions like Selman suggested when we discussed this in the past

File: src/components/projects/View.js [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+  }, [dispatch, page, pageSize, selectedProjectStatus, searchInput]);
+
+  const dataSource = useSelector((state) => state.projects.projects);
```
Comment (AntMarkoLovric) [inline src/components/projects/View.js]: when you move actions, you can move selector also and remove useSelector from component

File: src/components/projects/components/data.js
Comment (AntMarkoLovric) [inline src/components/projects/components/data.js:1]: why didn't you remove this if you don't use it?

File: src/components/projects/index.js
Comment (AntMarkoLovric) [inline src/components/projects/index.js:1]: this file is used to make connection between container and view.js

File: src/index.scss
Comment (AntMarkoLovric) [inline src/index.scss:1]: why did you change .css to .scss

File: src/reducers/projectsReducer.js
Comment (AntMarkoLovric) [inline src/reducers/projectsReducer.js:1]: project reducer goes into projects/modules/reducer, and with that approach we fulfill fractal folder structure

File: src/sagas/projectsSaga.js
Comment (AntMarkoLovric) [inline src/sagas/projectsSaga.js:1]: also move this to projects/modules/saga

File: src/sagas/projectsSaga.js [hunk 24]
```diff
@@ -0,0 +1,68 @@
...
+} from '../components/projects/modules/types';
+import {
+  fetchProjectsSuccess,
+  fetchProjectsFailure,
+  fetchProjectsRequest, 
+} from '../components/projects/modules/actions';
+
+
+function* fetchProjectsSaga(action) {
+  const { page, pageSize, selectedProjectStatus, searchInput } = action.payload;
```
Comment (AntMarkoLovric) [inline src/sagas/projectsSaga.js]: without deconstructing this, you could pass an object to service and with that you can reduce number of parameters because best function is one without parameters

File: src/sagas/projectsSaga.js [hunk 25]
```diff
@@ -0,0 +1,68 @@
...
+      page,
+      pageSize,
+      selectedProjectStatus,
+      searchInput
+    );
+
+    const projects = response.data.projects;
+    const total = response.data.count;
+
+    const formattedData = projects.map((project, index) => {
```
Comment (AntMarkoLovric) [inline src/sagas/projectsSaga.js]: can we reduce complexity of this function using just one map, by removing map inside of a map

File: src/sagas/projectsSaga.js [hunk 26]
```diff
@@ -0,0 +1,68 @@
...
+      const developers = project.developers.map((developer) => developer.user);
```
Comment (AntMarkoLovric) [inline src/sagas/projectsSaga.js]: you have two identical maps

File: src/sagas/projectsSaga.js [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+        hourlyRate: project.hourlyRate || 0,
+        projectValue: project.projectValue || 0,
+        status: project.projectStatus || '',
+      };
+    });
+
+    yield put(fetchProjectsSuccess(formattedData, total));
+  } catch (error) {
+    yield put(fetchProjectsFailure(error));
+  } finally {
```
Comment (AntMarkoLovric) [inline src/sagas/projectsSaga.js]: in "finally" block, we don't trigger fetch, we can, for example, turn off loader

**Review by AntMarkoLovric — CHANGES_REQUESTED** [review]

File: src/components/Table/CustomTable.js [hunk 28]
```diff
@@ -0,0 +1,92 @@
...
+  ) : null;
+
+  const tableContent = isLoading ? null : (
+    <Table 
+      dataSource={pageData}
+      columns={columns}
+      className="table"
+      pagination={false}
+      bordered
+      title={() => tableHeader}
```
Comment (becarselman) [inline src/components/Table/CustomTable.js]: Why you create function here. I think you can just pass table Header.

File: src/components/Table/components/Navbar.js [hunk 29]
```diff
@@ -0,0 +1,34 @@
...
+    handlePageSelect(index + 1, label);
+    setSelectedPage(index + 1);
+  };
+
+  const navLinks = navLabels.map((label, index) => (
+    <NavLink
+      key={index}
+      label={label}
+      onClick={() => handleClick(index, label)}
+      className={`${selectedPage === index + 1 ? 'active-link' : ''} ${
```
Comment (becarselman) [inline src/components/Table/components/Navbar.js]: use classnames instead of this

File: src/components/Table/modules/actions.js
Comment (becarselman) [inline src/components/Table/modules/actions.js:1]: I can't see anywhere you use this actions. How I see you create this and leave old useState.

File: src/components/Table/CustomTable.js [hunk 30]
```diff
@@ -0,0 +1,92 @@
...
+  const handlePageChange = (page) => {
+    onPageChange(page);
+  };
+
+  const handleFilterChange = (label) => {
+    setFilter(label);
+  };
+
+  useEffect(() => {
+    const filteredData = data.filter((item) => {
```
Comment (becarselman) [inline src/components/Table/CustomTable.js]: you need to do this on backend. To be sure you get right data you will need to filter them on backend

File: src/components/Table/components/Navbar.js [hunk 31]
```diff
@@ -0,0 +1,34 @@
...
+  const handleClick = (index, label) => {
+    handlePageSelect(index + 1, label);
+    setSelectedPage(index + 1);
+  };
+
+  const navLinks = navLabels.map((label, index) => (
+    <NavLink
+      key={index}
+      label={label}
+      onClick={() => handleClick(index, label)}
```
Comment (becarselman) [inline src/components/Table/components/Navbar.js]: move this function outside of jsx and don't use inline function.

File: src/components/Table/components/TableHeader.js [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <div className="table-header">
+      <div className="header-content">
+        <h2 className="header-title">All Projects</h2>
+        <div className="project-count">{totalCount} total</div>
+      </div>
+      <form
+        className="search-form"
+        onSubmit={(event) => {
```
Comment (becarselman) [inline src/components/Table/components/TableHeader.js]: move this on submit outside of jsx. this Whole block can be part of some function

File: src/components/Table/components/TableHeader.js [hunk 33]
```diff
@@ -0,0 +1,34 @@
...
+          event.preventDefault();
+          handleSearch(input);
+        }}
+      >
+        <Input
+          className="search-input"
+          placeholder="Search"
+          value={input}
+          onChange={(event) => setInput(event.target.value)}
```
Comment (becarselman) [inline src/components/Table/components/TableHeader.js]: same

File: src/components/projects/components/columns.js [hunk 34]
```diff
@@ -0,0 +1,55 @@
...
+      },
+    },
+    {
+      title: 'Status',
+      dataIndex: 'status',
+      key: 'status',
+      render: (projectStatus) => (
+        <span>
+          <span
+           className={`status-dot ${projectStatus === 'Active' ?
```
Comment (becarselman) [inline src/components/projects/components/columns.js]: use classNames here  https://www.npmjs.com/package/classnames

**Review by becarselman — CHANGES_REQUESTED** [review]

File: src/components/Table/CustomTable.js [hunk 35]
```diff
@@ -0,0 +1,110 @@
...
+      return item.status === filter || filter === 'All Projects';
+    });
+    setPageDataState(filteredData);
+  }, [data, filter]);
+
+  useEffect(() => {
```
Comment (becarselman) [inline src/components/Table/CustomTable.js]: Why you use this in use effect. Why you don't set onClick

File: src/components/Table/components/TableHeader.js [hunk 36]
```diff
@@ -0,0 +1,54 @@
...
+  const [input, setInput] = useState('');
+  const [countChanged, setCountChanged] = useState(false);
+
+  useEffect(() => {
+    setInput(searchValue);
+  }, [searchValue]);
+
+  useEffect(() => {
+    setCountChanged(true);
+    const timeout = setTimeout(() => {
```
Comment (becarselman) [inline src/components/Table/components/TableHeader.js]: why you use timeout. Please find better solution.

File: src/components/projects/View.js [hunk 37]
```diff
@@ -0,0 +1,64 @@
...
+    setSearchInput(input);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [page, pageSize, selectedProjectStatus, searchInput]);
+
+  const fetchData = () => {
+    actions.fetchAllProjects(page, pageSize, selectedProjectStatus, searchInput);
+    dispatch(setPageData(null));
```
Comment (becarselman) [inline src/components/projects/View.js]: Don't use dispatch

File: src/components/projects/View.js [hunk 38]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  const handleNavSelect = (label) => {
+    setSelectedProjectStatus(label);
+    setPage(1);
+    dispatch(setPageData(null));
```
Comment (becarselman) [inline src/components/projects/View.js]: same

File: src/components/sidebar/Sidebar.js [hunk 39]
```diff
@@ -1,51 +1,73 @@
...
+
+  const arrayOfDivsSidebarOptions = sidebarOptions.map((option) => {
+    const isActive = option.path === location.pathname;
     return (
-      <div className="sidebar-option" key={option.text}>
+      <Link
+        to={option.path}
+        className={`sidebar-option ${isActive ? "active" : ""}`}
+        key={option.text}
+        onClick={() => handleClick(option.text)}
```
Comment (becarselman) [inline src/components/sidebar/Sidebar.js]: move out of jsx

File: src/components/sidebar/Sidebar.scss [hunk 40]
```diff
@@ -1,22 +1,26 @@
...
 
   background: linear-gradient(179.32deg, $background-gradient-first-color 8.04%, $background-gradient-second-color 99.41%);
   border-right: 1px solid $border-color;
 
-  height: 100vh;
+  max-height: 100vh;
+  height: 100%;
 
-  padding: 30px;
+  // max-width: 325px;
```
Comment (becarselman) [inline src/components/sidebar/Sidebar.scss]: remove commented code

File: src/components/sidebar/Sidebar.scss [hunk 41]
```diff
@@ -97,4 +100,53 @@
...
+    opacity: 0;
+    transform: scale(1.2);
+  }
+}
+
+.dashboard-page {
+  flex: 1;
+  background-color: #fff;
+  margin-left: 250px;
+  /* Dodati sledeći stil za poplavljivanje dužine */
```
Comment (becarselman) [inline src/components/sidebar/Sidebar.scss]: use only english

File: src/components/Table/components/Navbar.js [hunk 42]
```diff
@@ -0,0 +1,43 @@
...
+      'last-link': index === navLabels.length - 1
+    });
+
+    const handleNavLinkClick = () => handleClick(index, label);
+
+    return (
+      <NavLink
+        key={index}
+        label={label}
+        onClick={handleNavLinkClick}
```
Comment (becarselman) [inline src/components/Table/components/Navbar.js]: why you don't pass handle click instead of creating two functions

**Review by becarselman — CHANGES_REQUESTED** [review]

File: src/components/Table/CustomTable.js [hunk 43]
```diff
@@ -0,0 +1,110 @@
...
+  const handleFilterChange = (label) => {
+    setFilter(label);
+  };
+
+  const handleSearchValueChange = (value) => {
+    setSearchValue(value);
+  };
+
+  const handleClick = () => {
+    const filteredData = data.filter((item) => {
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.js]: why you filter data if we get filtered data from backend

File: src/components/Table/components/Navbar.js [hunk 44]
```diff
@@ -0,0 +1,39 @@
...
+      'active-link': selectedPage === index + 1,
+      'first-link': index === 0,
+      'last-link': index === navLabels.length - 1
+    });
+
+    return (
+      <NavLink
+        key={index}
+        label={label}
+        onClick={() => handleClick(index, label)}
```
Comment (AntMarkoLovric) [inline src/components/Table/components/Navbar.js]: move onClick outside of jsx

File: src/components/projects/View.js [hunk 45]
```diff
@@ -0,0 +1,61 @@
...
+import { setPageData } from './modules/actions';
+
+const Projects = ({
+  dataSource,
+  totalCount,
+  isLoading,
+  actions,
+}) => {
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
```
Comment (AntMarkoLovric) [inline src/components/projects/View.js]: This needs to be part of table component because it will be same for all tables

File: src/sagas/authSaga.js [hunk 46]
```diff
@@ -12,15 +12,27 @@ function* loginSaga(action) {
...
-    yield put(loginError(error.response.data)); 
-    yield put(getErrors(error.response.data)); 
-  } 
+    yield put(loginError(error.response.data));
+    yield put(getErrors(error.response.data));
+  }
 }
 
-
+function* logoutSaga() {
```
Comment (AntMarkoLovric) [inline src/sagas/authSaga.js]: this is not in your task, what did you do here?

File: src/service/projectsService.js [hunk 47]
```diff
@@ -0,0 +1,15 @@
+import axiosInstance from './apiService';
+const URL = process.env.REACT_APP_API_URL;
+
+export const getAllProjects = ({ page, pageSize, selectedProjectStatus, searchInput }) => {
+  const statusQueryParam = selectedProjectStatus !== 'All Projects' ? selectedProjectStatus : '';
```
Comment (AntMarkoLovric) [inline src/service/projectsService.js]: can we do this in saga?

File: src/service/projectsService.js [hunk 48]
```diff
@@ -0,0 +1,15 @@
...
+  const searchQueryParam = searchInput ? `search[name]=${encodeURIComponent(searchInput)}` : '';
```
Comment (AntMarkoLovric) [inline src/service/projectsService.js]: same as the comment above

File: src/components/Table/CustomTable.scss [hunk 49]
```diff
@@ -0,0 +1,212 @@
...
+.ant-pagination-next, .ant-pagination-prev {
+  display: none !important;
+}
+
+.ant-table-row {
+    transition: transform 0.3s;
+    will-change: transform;
+    cursor: pointer;
+  
+    &:hover {
```
Comment (AntMarkoLovric) [inline src/components/Table/CustomTable.scss]: i know you have good intention with using on hover, but this doesn't look good

**Review by AntMarkoLovric — CHANGES_REQUESTED** [review]

File: src/components/Table/CustomTable.scss [hunk 50]
```diff
@@ -0,0 +1,222 @@
...
+.pagination-container {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  width: 100%;
+  margin-top: 16px;
+
+  .pagination-info {
+    @include gilroy-medium500;
+    color: #6C6D75;
```
Comment (becarselman) [inline src/components/Table/CustomTable.scss]: move color into colors file

File: src/components/Table/components/Navlink.js [hunk 51]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import './Navlink.scss';
+
+const NavLink = ({ label, onClick, className }) => {
+  const handleClick = () => {
```
Comment (becarselman) [inline src/components/Table/components/Navlink.js]: You don't need this function you can use onClick in jsx as well. this function just call function from props. Why you don't use directly

File: src/components/Table/components/TableHeader.js [hunk 52]
```diff
@@ -0,0 +1,45 @@
...
+
+  const handleSearchSubmit = (event) => {
+    event.preventDefault();
+    handleSearch(searchValue);
+  };
+
+  return (
+    <div className="table-header">
+      <div className="header-content">
+        <h2 className="header-title">All Projects</h2>
```
Comment (becarselman) [inline src/components/Table/components/TableHeader.js]: Because we need to reuse this we can not use All Projects here. we need to get this title from props. And then it will be dynamic. If we go to employees page we want to se all employees instead of project. Am I right

File: src/components/Table/components/TableHeader.scss [hunk 53]
```diff
@@ -0,0 +1,85 @@
...
+
+    .project-count {
+      @include gilroy-medium500();
+      display: flex;
+      height: 1.875rem;
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      border-radius: 0.25rem;
+      padding: 0.625rem;
```
Comment (becarselman) [inline src/components/Table/components/TableHeader.scss]: is this 10px?. This is so hard to read.

File: src/components/Table/components/TableHeader.scss [hunk 54]
```diff
@@ -0,0 +1,85 @@
...
+      font-size: 0.875rem;
```
Comment (becarselman) [inline src/components/Table/components/TableHeader.scss]: 14px?

File: src/components/Table/components/TableHeader.scss (hunk 53 above)
Comment (becarselman) [reply src/components/Table/components/TableHeader.scss]: And also I think this is not to much important to be in relative units. We can fix this to suite design

File: src/components/projects/View.js [hunk 55]
```diff
@@ -0,0 +1,55 @@
...
+}) => {
+  const [selectedProjectStatus, setSelectedProjectStatus] = useState('All Projects');
+  const [searchInput, setSearchInput] = useState('');
+
+  const handleSearchChange = (input) => {
+    setSearchInput(input);
+  };
+
+  useEffect(() => {
+    fetchData(1, 10);
```
Comment (becarselman) [inline src/components/projects/View.js:21]: Leave this for now. But I think this will become really messy in future with all this numbers. My personal opinion is to reset page number and page size in redux state. But for now leave

File: src/components/projects/modules/saga.js [hunk 56]
```diff
@@ -0,0 +1,68 @@
+import { takeLatest, put, call } from 'redux-saga/effects';
+import { fetchAllProjectsSuccess, fetchAllProjectsFailure, startLoader, stopLoader } from './actions';
+import { actionTypes } from "./types";
+import { getAllProjects } from '../../../service/projectsService';
+
+function generateQueryParams(selectedProjectStatus, searchInput) {
+  const statusQueryParam = selectedProjectStatus !== 'All Projects' ? selectedProjectStatus : '';
+  const searchQueryParam = searchInput ? `search[name]=${encodeURIComponent(searchInput)}` : '';
```
Comment (becarselman) [inline src/components/projects/modules/saga.js]: Why you encode this. If user type name of project its visible already. He don't send anything private or important. he just type search

**Review by becarselman — CHANGES_REQUESTED** [review]

**Review by becarselman — APPROVED** [review]

**Review by ElvaRivle — APPROVED** [review]
